nomos-mix/core/src/message_blend/mod.rs

```diff
@@ -153,7 +58,9 @@ where
         let (bypass_sender, bypass_receiver) = mpsc::unbounded_channel();
         let (temporal_sender, temporal_receiver) = mpsc::unbounded_channel();
         let output_stream = tokio_stream::StreamExt::merge(
-            UnboundedReceiverStream::new(bypass_receiver),
+            UnboundedReceiverStream::new(bypass_receiver)
+                .persistent_transmission(settings.persistent_transmission)
```

Comment:
This is where the magic happens. Please check that correct messages are forwarded to the persisten transmission. But I think it should be ok.